pkg/gather/clusterconfig/sap_config.go

```diff
@@ -0,0 +1,113 @@
+package clusterconfig
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/kubernetes"
+	corev1client "k8s.io/client-go/kubernetes/typed/core/v1"
+
+	authclient "github.com/openshift/client-go/authorization/clientset/versioned/typed/authorization/v1"
+	securityv1client "github.com/openshift/client-go/security/clientset/versioned/typed/security/v1"
+
+	"github.com/openshift/insights-operator/pkg/record"
+)
+
+// GatherSAPConfig collects selected security context constraints
+// and cluster role bindings from clusters running a SAP payload.
+//
+// Relevant OpenShift API docs:
+//   - https://pkg.go.dev/github.com/openshift/client-go/authorization/clientset/versioned/typed/authorization/v1
+//   - https://pkg.go.dev/github.com/openshift/client-go/security/clientset/versioned/typed/security/v1
+//
+// Location in archive: config/sap_namespaces.json, config/securitycontentconstraint/, config/clusterrolebinding/
+func GatherSAPConfig(g *Gatherer, c chan<- gatherResult) {
+	gatherDynamicClient, err := dynamic.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		c <- gatherResult{errors: []error{err}}
+		return
+	}
+	gatherKubeClient, err := kubernetes.NewForConfig(g.gatherProtoKubeConfig)
+	if err != nil {
+		c <- gatherResult{errors: []error{err}}
+		return
+	}
+	gatherSecurityClient, err := securityv1client.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		c <- gatherResult{errors: []error{err}}
+		return
+	}
+	gatherAuthClient, err := authclient.NewForConfig(g.gatherKubeConfig)
+	if err != nil {
+		c <- gatherResult{errors: []error{err}}
+		return
+	}
+
+	records, errors := gatherSAPConfig(g.ctx, gatherDynamicClient, gatherKubeClient.CoreV1(), gatherSecurityClient, gatherAuthClient)
+	c <- gatherResult{records: records, errors: errors}
+}
+
+func gatherSAPConfig(ctx context.Context, dynamicClient dynamic.Interface, coreClient corev1client.CoreV1Interface, securityClient securityv1client.SecurityV1Interface, authClient authclient.AuthorizationV1Interface) ([]record.Record, []error) {
+	sccFilter := map[string]struct{}{"anyuid": {}, "privileged": {}}
+	crbFilter := map[string]struct{}{"system:openshift:scc:anyuid": {}, "system:openshift:scc:privileged": {}}
```

Comment:
Do we need maps for these? Wouldn't be array of string sufficient?

Reply:
You're right. It would work fine, but this syntax ensures that each resource is collected at most once. Originally, there was a `name` field check when iterating over a list of returned results instead of using `Get`, which checked if the name existed as a key in this map (which is actually the canonical Go syntax for a set) because it's faster and shorter than iterating over a list. I can change it if you want, but I think this should be fairly clear to anyone aware of this somewhat unusual syntax for sets).